src/api/structure/testing-unit.js

```diff
@@ -48,11 +49,12 @@ export default class TestingUnit {
 
         assertType(is.string, 'page', 'The page URL', this.pageUrl);
 
-        if (!PROTOCOL_RE.test(this.pageUrl) && this.pageUrl !== 'about:blank') {
-            var protocol = IMPLICIT_PROTOCOL_RE.test(this.pageUrl) ? 'http:' : 'http://';
+        var urlValidation = validateUrl(this.pageUrl);
 
-            this.pageUrl = protocol + this.pageUrl;
-        }
+        if (!urlValidation.valid && urlValidation.reason.protocol)
+            throw new GeneralError(MESSAGES.unsupportedUrlProtocol, this.pageUrl, urlValidation.reason.protocol);
+
+        this.pageUrl = buildTestPageUrl(this.pageUrl, this.testFile.filename);
```

Comment:
@AndreyBelym Then it should be named `resolvePageUrl` or smthg like that. BTW, it shouldn't be placed in `utils` dir, since it's not about some generic URL, it's specific to URLs in API